server/block/cactus.go

```diff
@@ -25,33 +25,20 @@ func (c Cactus) UseOnBlock(pos cube.Pos, face cube.Face, _ mgl64.Vec3, w *world.
 	if !used {
 		return false
 	}
-	_, ok := w.Block(pos.Side(cube.FaceDown)).(Cactus)
-	if !supportsVegetation(c, w.Block(pos.Side(cube.FaceDown))) && !ok {
+	if !c.canGrowHere(pos, w, 2) {
```

Comment:
This was missed in the previous review but the total cactus height should not be considered here either.

Reply:
height excluded from check.